### PR TITLE
catch and record ssl, dns related errors in the http probe

### PR DESCRIPTION
**Pull request by pranj**

### CODE REVIEW

File: probes/http/http.go [hunk 1]
```diff
@@ -72,11 +73,11 @@ type Probe struct {
 }
 
 type result struct {
-	total, success, timeouts int64
-	latency                  metrics.Value
-	respCodes                *metrics.Map
-	respBodies               *metrics.Map
-	validationFailure        *metrics.Map
+	total, success, timeouts, dnserrors, sslerrors int64
```
Comment (eefi) [inline probes/http/http.go]: Should these be dnsErrors and sslErrors?

File: probes/http/http.go [hunk 2]
```diff
@@ -237,6 +265,16 @@ func (p *Probe) doHTTPRequest(req *http.Request, result *result) {
 			result.timeouts++
 			return
 		}
+		if isDNSError(err) {
+			p.l.Warning("Target:", req.Host, ", URL:", req.URL.String(), ", http.doHTTPRequest: dns error: ", err.Error())
```
Comment (eefi) [inline probes/http/http.go]: Should the text say "DNS" instead of "dns", since it is technically an acronym?

File: probes/http/http.go [hunk 3]
```diff
@@ -237,6 +265,16 @@ func (p *Probe) doHTTPRequest(req *http.Request, result *result) {
...
+			result.dnserrors++
+			return
+		}
+		if isSSLError(err) {
+			p.l.Warning("Target:", req.Host, ", URL:", req.URL.String(), ", http.doHTTPRequest: ssl error: ", err.Error())
```
Comment (eefi) [inline probes/http/http.go]: Should the text say "SSL" instead of "ssl"/

**Review by eefi — COMMENTED** [review]
Overall, the approach LGTM. A couple minor capitalization suggestions.

**Comment by pranj** [comment]
Done

**Review by eefi — APPROVED** [review]
LGTM!

**Pull request merged**